cmake/wayland_driver.cmake

```diff
@@ -0,0 +1,61 @@
+project(lv_wayland)
+
+option(USE_WL_SHELL "use wl shell (discard, no recommend)" OFF)
+option(USE_XDG_SHELL "use xdg shell" OFF)
+option(USE_WAYLAND_TIMER_HANDLER "" OFF)
+
+set(WAYLAND_SCANNER_EXECUTABLE)
+
+macro(wayland_generate protocol_xml_file output_dir)
+
+    if(NOT EXISTS ${protocol_xml_file})
+        message(FATAL_ERROR "-- protocol XML file not found: " ${protocol_xml_file})
+    endif()
+
+    get_filename_component(output_file_base ${protocol_xml_file} NAME_WE)
+    set(xdg_shell_file "${output_dir}/wayland-${output_file_base}-client-protocol")
+    message("-- generating ${xdg_shell_file}.h")
+    execute_process(COMMAND "${WAYLAND_SCANNER_EXECUTABLE}"  client-header  "${protocol_xml_file}" "${xdg_shell_file}.h")
+    message("-- generating ${xdg_shell_file}.c")
+    execute_process(COMMAND "${WAYLAND_SCANNER_EXECUTABLE}"  private-code "${protocol_xml_file}"  "${xdg_shell_file}.c")
+endmacro()
+
+if (NOT USE_WL_SHELL AND USE_XDG_SHELL)
```

Comment:
Here you have a bug, it should be:

`if (NOT USE_WL_SHELL AND NOT USE_XDG_SHELL)`

Otherwise, enabling only XDG shell would trigger the FATAL_ERROR

Reply:
modified already